### PR TITLE
Add shopping cart, cart item and functionality for them

### PR DESCRIPTION
**Pull request by Den-moroz**

### CODE REVIEW

File: src/main/java/core/basesyntax/bookstore/service/impl/CartItemServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+public class CartItemServiceImpl implements CartItemService {
+    private final UserService userService;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    @Transactional
+    public CartItemDto save(CreateCartItemDto cartItemDto) {
```
Comment (andrii-hoienko) [inline src/main/java/core/basesyntax/bookstore/service/impl/CartItemServiceImpl.java:34]: There are too many responsibilities for one method here, please make this more SOLID

File: src/main/java/core/basesyntax/bookstore/service/UserService.java [hunk 2]
```diff
@@ -3,8 +3,11 @@
 import core.basesyntax.bookstore.dto.user.UserRegistrationRequestDto;
 import core.basesyntax.bookstore.dto.user.UserRegistrationResponseDto;
 import core.basesyntax.bookstore.exception.RegistrationException;
+import core.basesyntax.bookstore.model.User;
 
 public interface UserService {
     UserRegistrationResponseDto register(UserRegistrationRequestDto request)
             throws RegistrationException;
+
+    User getUser();
```
Comment (andrii-hoienko) [inline src/main/java/core/basesyntax/bookstore/service/UserService.java]: What user do you get? It is not clear from method name

File: src/main/resources/db/changelog/changes/08-create-cart-item-table.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-cart-item-table
+      author: denis
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/08-create-cart-item-table.yaml:4]: What if there are 2 denis in team?

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, some minor improvements required

File: src/main/java/core/basesyntax/bookstore/model/ShoppingCart.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    @JoinColumn(name = "user_id", nullable = false)
+    private User user;
+    @ManyToMany
+    @JoinTable(name = "shopping_cart_cart_item",
+            joinColumns = @JoinColumn(name = "shopping_cart_id"),
+            inverseJoinColumns = @JoinColumn(name = "cart_item_id")
+    )
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    private List<CartItem> cartItems;
```
Comment (vovasalyha) [inline src/main/java/core/basesyntax/bookstore/model/ShoppingCart.java:38]: It's better to initialize collections here, so Hibernate can choose a right implementation of proxied collection

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**